File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
import java.util.*;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int len = 0, i = 0, j = 0;
        HashSet<Character> hs = new HashSet<>();

        while (j < s.length()) {
            if (hs.contains(s.charAt(j))) {
                hs.remove(s.charAt(i));
                i++;
            } else {
                hs.add(s.charAt(j));
                len = Math.max(len, j - i + 1);
                j++;
            }
        }

        return len;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int low =0;
        int high =0;
        unordered_map<char,int> f;
        int res = 0;
        int n = s.size();


        for(int high =0 ; high <n; high++){
            f[s[high]]++;

             int k = high - low +1;
            while(f[s[high]]>1){
                f[s[low]]--;
                if(f[s[low]]==0){f.erase(s[low]);}

              
low++;

            }
              
            int len = high - low +1;
            res = max(res, len);
        }
        return res;
    }
};